prov/rxm/src/rxm_conn.c

```diff
@@ -613,9 +612,7 @@ int rxm_cmap_process_connreq(struct rxm_cmap *cmap, void *addr,
 		} else {
 			FI_DBG(cmap->av->prov, FI_LOG_EP_CTRL,
 				"Endpoint connects to itself\n");
-			ret = rxm_cmap_alloc_handle_peer(cmap, addr,
-							  RXM_CMAP_CONNREQ_RECV,
-							  &handle);
+			ret = rxm_cmap_alloc_handle_peer(cmap, addr, &handle);
```

Comment:
Thanks. I was setting the handle state after the check for the bad return since the handle is not allocated in that case. I was attempting to follow the same logic as the original. Looking back at this now though, the RXM_CM_UPDATE_STATE(handle, RXM_CMAP_CONNREQ_RECV); that follows the error check in the else clause is redundant. That code will fall through to the next case which will set it. I will remove the extra set. Let me know if I addressed the right area in the code or further clarification is needed.